src/pages/ExternalSupplierInvoicePage.js

```diff
@@ -44,80 +45,36 @@ const SORT_DATA_TYPES = {
   packSize: 'number',
 };
 
-export class ExternalSupplierInvoicePage extends GenericPage {
+export class ExternalSupplierInvoicePage extends React.Component {
   constructor(props) {
     super(props);
-    this.state.sortBy = 'itemName';
-    this.state.totalPrice = 0;
-    this.state.expiryModelIsOpen = false;
-    this.state.columns = [
-      {
-        key: 'itemCode',
-        width: 2,
-        title: tableStrings.item_code,
-        sortable: true,
-      },
-      {
-        key: 'itemName',
-        width: 4,
-        title: tableStrings.item_name,
-        sortable: true,
-      },
-      {
-        key: 'packSize',
-        width: 1.3,
-        title: tableStrings.pack_size,
-        alignText: 'center',
-      },
-      {
-        key: 'numberOfPacks',
-        width: 1.3,
-        title: tableStrings.quantity,
-        alignText: 'center',
-      },
-      {
-        key: 'batch',
-        width: 1.8,
-        title: tableStrings.batch_name,
-        alignText: 'center',
-      },
-      {
-        key: 'expiryDate',
-        width: 2,
-        title: tableStrings.batch_expiry,
-        alignText: 'center',
-      },
-      {
-        key: 'costPrice',
-        width: 1.4,
-        title: tableStrings.batch_cost_price,
-        alignText: 'center',
-      },
-      {
-        key: 'remove',
-        width: 1,
-        title: tableStrings.remove,
-        alignText: 'center',
-      },
-    ];
-    this.dataTypesSynchronised = DATA_TYPES_SYNCHRONISED;
-    this.finalisableDataType = 'Transaction';
-    this.getFilteredSortedData = this.getFilteredSortedData.bind(this);
-    this.onEndEditing = this.onEndEditing.bind(this);
-    this.onDatabaseEvent = this.onDatabaseEvent.bind(this);
-    this.openItemSelector = this.openItemSelector.bind(this);
-    this.openCommentEditor = this.openCommentEditor.bind(this);
-    this.openTheirRefEditor = this.openTheirRefEditor.bind(this);
-    this.getModalTitle = this.getModalTitle.bind(this);
-    this.addNewLine = this.addNewLine.bind(this);
+    this.dataFilters = {
+      searchKey: '',
+      sortBy: 'itemName',
+      isAscending: true,
+    };
+    this.state = {
+      totalPrice: 0,
+      modalKey: null,
+      modalIsOpen: false,
+      selection: [],
+    };
+    autobind(this);
   }
+
+  updateDataFilters(newSearchTerm, newSortBy, newIsAscending) {
```

Comment:
Concept I haven't though of - keeping things out of state that shouldn't trigger component updates (potentially saving time on shouldComponentUpdate).